src/ts/components/settingspanel.ts

```diff
@@ -152,11 +294,21 @@ export class SettingsPanel extends Container<SettingsPanelConfig> {
 
   /**
    * Gets the event that is fired when one or more {@link SettingsPanelItem items} have changed state.
+   * @deprecated
    * @returns {Event<SettingsPanel, NoArgs>}
    */
   get onSettingsStateChanged(): Event<SettingsPanel, NoArgs> {
     return this.settingsPanelEvents.onSettingsStateChanged.getEvent();
   }
+
+  addComponent(component: Component<ComponentConfig>) {
+    if (component instanceof SettingsPanelPage) {
+      super.addComponent(component);
+    } else {
+      // backwards compatibility
+      this.getRootPage().addComponent(component);
```

Comment:
It must not be anything other than a `SettingsPanelPage`. Remove the backwards compatibility, not necessary for v3.